sim/aviron/src/lib/io.zig

```diff
@@ -191,3 +161,32 @@ pub const IO = struct {
         return null;
     }
 };
+
+// Unified byte-addressable device interface used by MemorySpace for RAM and IO
+pub const Device = struct {
+    ctx: *anyopaque,
+    vtable: *const VTable,
+
+    pub const VTable = struct {
+        read8: *const fn (ctx: *anyopaque, idx: usize) u8,
+        write8: *const fn (ctx: *anyopaque, idx: usize, v: u8) void,
+        write_masked: *const fn (ctx: *anyopaque, idx: usize, mask: u8, v: u8) void,
```

Comment:
I would use `read` and `write` instead of `read8` and `write8`. Will also require less changes on the whole codebase and doesn't convey any meaningful information as all `Bus` devices will have 8 bit access only anyways